sgp30/Makefile

```diff
@@ -1,5 +1,6 @@
 sensirion_common_dir := ../embedded-common
 sgp_common_dir := ../sgp-common
+platform ?= sample-implementations/linux_user_space
```

Comment:
I think the generic platform should be the stub, but if we recognize the native or the CROSS_COMPILE platform we could apply a mapping..

Reply:
The thing is it doesn't even work for most of our targets, since some like Arduino actually use a cpp file.

Reply:
ok, the dir doesn't work, but if we use the file path (dir + sourcefile), then this could work, right)?

Then we'd need some kind of dict to catch the platform
```
platform_swi2c_linux := sw_i2c/sample-implementations/linux_user_space/sensirion_sw_i2c_implementation.c
platform_swi2c_arduino := sw_i2c/sample-implementations/arduino/sensirion_sw_i2c_implementation.cpp

platform := $(platform_swi2c_$(shell magic))
```

I'm not saying it's the way to go though - it seems too complex for what we're trying to achieve..